main.tf

```diff
@@ -0,0 +1,112 @@
+  source     = "git::https://github.com/cloudposse/tf_label.git?ref=tags/0.2.0"
+  namespace  = "${var.namespace}"
+  stage      = "${var.stage}"
+  name       = "${var.name}"
+  delimiter  = "${var.delimiter}"
+  attributes = ["origin"]
+  tags       = "${var.tags}"
+}
+
+resource "aws_cloudfront_origin_access_identity" "default" {
+  comment = "${module.distribution_label.id}"
+}
+
+module "logs" {
+  source                   = "git::https://github.com/cloudposse/tf_log_storage.git?ref=tags/0.1.0"
+  namespace                = "${var.namespace}"
+  stage                    = "${var.stage}"
+  name                     = "${var.name}"
+  delimiter                = "${var.delimiter}"
+  attributes               = ["logs"]
+  tags                     = "${var.tags}"
+  prefix                   = "${var.log_prefix}"
+  standard_transition_days = "${var.log_standard_transition_days}"
+  glacier_transition_days  = "${var.log_glacier_transition_days}"
+  expiration_days          = "${var.log_expiration_days}"
+}
+
+module "distribution_label" {
+  source    = "git::https://github.com/cloudposse/tf_label.git?ref=tags/0.2.0"
+  namespace = "${var.namespace}"
+  stage     = "${var.stage}"
+  name      = "${var.name}"
+  delimiter = "${var.delimiter}"
+  tags      = "${var.tags}"
+}
+
+resource "aws_cloudfront_distribution" "default" {
+  enabled             = "${var.enabled}"
+  is_ipv6_enabled     = "${var.is_ipv6_enabled}"
+  comment             = "${var.comment}"
+  default_root_object = "${var.default_root_object}"
+  price_class         = "${var.price_class}"
+
+  logging_config = {
+    include_cookies = "${var.log_include_cookies}"
+    bucket          = "${module.logs.bucket_domain_name}"
+    prefix          = "${var.log_prefix}"
+  }
+
+  aliases = "${var.aliases}"
+
+  origin {
+    domain_name = "${var.origin_domain_name}"
+    origin_id   = "${module.distribution_label.id}"
+    origin_path = "${var.origin_path}"
+
+    custom_origin_config {
+      http_port                = "${var.origin_http_port}"
+      https_port               = "${var.origin_https_port}"
+      origin_protocol_policy   = "${var.origin_protocol_policy}"
+      origin_ssl_protocols     = "${var.origin_ssl_protocols}"
+      origin_keepalive_timeout = "${var.origin_keepalive_timeout}"
+      origin_read_timeout      = "${var.origin_read_timeout}"
+    }
+  }
+
+  viewer_certificate {
+    acm_certificate_arn            = "${var.acm_certificate_arn}"
+    ssl_support_method             = "sni-only"
+    minimum_protocol_version       = "TLSv1"
+    cloudfront_default_certificate = true
+  }
+
+  default_cache_behavior {
+    allowed_methods  = "${var.allowed_methods}"
+    cached_methods   = "${var.cached_methods}"
+    target_origin_id = "${module.distribution_label.id}"
+    compress         = "${var.compress}"
+
+    forwarded_values {
+      query_string = "${var.forward_query_string}"
+
+      cookies {
+        forward = "${var.forward_cookies}"
+      }
+    }
```

Comment:
In many cases when having a website behind a `CloudFront` CDN, we need different URL paths to use different cache behaviors, _e.g._ `/` would use one behavior while `/images/*` would use another (probably with different  TTLs). How could we provide additional cache behavior if we need them?

Reply:
That's not addressed by this module. CloudFront respect's the origin's cache headers, so that's the best way to support it. Until we're faced by a use-case not addressed by this workaround, we'll leave it out.